Jellyfin.Server/Migrations/JellyfinMigrationBackupAttribute.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+
+namespace Jellyfin.Server.Migrations;
+
+/// <summary>
+/// Marks an <see cref="JellyfinMigrationAttribute"/> migration and instructs the <see cref="JellyfinMigrationService"/> to perform a backup.
+/// </summary>
+[AttributeUsage(System.AttributeTargets.Class, Inherited = true, AllowMultiple = true)]
+public sealed class JellyfinMigrationBackupAttribute : System.Attribute
+{
+    /// <summary>
+    /// Gets or Sets a value indicating whether a backup of the old library.db should be performed.
+    /// </summary>
+    public bool LegacyLibraryDb { get; set; }
+
+    /// <summary>
+    /// Gets or Sets a value indicating whether a backup of the old library.db should be performed.
+    /// </summary>
+    public bool JellyfinDb { get; set; }
+
+    /// <summary>
+    /// Gets or Sets a value indicating whether a backup of the metadata folder should be performed.
+    /// </summary>
+    public bool Metadata { get; set; }
+
+    /// <summary>
+    /// Gets or Sets a value indicating whether a backup of the Trickplay folder should be performed.
+    /// </summary>
+    public bool Trickplay { get; set; }
+
+    /// <summary>
+    /// Gets or Sets a value indicating whether a backup of the Subtitles folder should be performed.
```

Comment:
Do we also want to take the attachment folder? If we're doing trickplay and subtitles?

Reply:
The backup service only offers those folders. Would need to extend that first

Reply:
![image](https://github.com/user-attachments/assets/bc69ded7-2fd1-4c61-8d5a-d294306ca462)

Yea i keep that in mind for a future extension